core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -201,8 +405,206 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
         builder.endObject();
     }
 
+    /**
+     * Returns the name of the query parser to use when parsing
+     * {@link SimpleQueryStringBuilder} objects from JSON.
+     */
     @Override
     public String queryId() {
         return NAME;
     }
+
+    /**
+     * Reads a {@link SimpleQueryStringBuilder} object from the given input
+     * stream.
+     * 
+     * @param in Stream to read the object from.
+     * @return the object read from the stream.
+     * */
+    @Override
+    public SimpleQueryStringBuilder readFrom(StreamInput in) throws IOException {
+        SimpleQueryStringBuilder result = new SimpleQueryStringBuilder(in.readString());
+        result.boost(in.readFloat());
+        int size = in.readInt();
+        Map<String, Float> fields = new HashMap<>();
+        for (int i = 0; i < size; i++) {
+            String field = in.readString();
+            Float weight = in.readFloat();
+            fields.put(field, weight);
+        }
+        result.fields(fields);
+
+        result.flags(in.readInt());
+        result.analyzer(in.readOptionalString());
+
+        result.defaultOperator(Operator.parseFromInt(in.readInt()));
+        result.lowercaseExpandedTerms(in.readBoolean());
+        result.lenient(in.readBoolean());
+        result.analyzeWildcard(in.readBoolean());
+
+        String localeStr = in.readString();
+        result.locale(Locale.forLanguageTag(localeStr));
+
+        result.queryName(in.readOptionalString());
+        result.minimumShouldMatch(in.readOptionalString());
+
```

Comment:
I think we prefer to not use setters here in general (I will document this), to be aligned with writeTo that doesn't use getters. just do e.g. result.queryName = in.readOptionalString();